cdap-cli/src/main/java/io/cdap/cdap/cli/CLIConfig.java

```diff
@@ -196,10 +196,6 @@ private UserAccessToken acquireAccessToken(
     ClientConfig clientConfig, ConnectionConfig connectionInfo, PrintStream output,
     boolean debug) throws IOException, UnauthorizedException {
 
-    if (!isAuthenticationEnabled(connectionInfo)) {
-      return null;
-    }
-
     try {
       UserAccessToken savedToken = getSavedAccessToken(connectionInfo.getHostname());
```

Comment:
So access token is always needed now??

Reply:
If the access token is saved, it is used before checking for authentication enabled. If not, check for authentication enabled is made. The authentication enabled check looks for a HTTP_UNAUTHORIZED (401) response which is not returned if there is a redirect for authentication.

Reply:
I see.. this throwing and exception and then catching it back in the same block and falling through is not very good code. Also, when savedToken is `null`, do we check for connection as well?

Reply:
Oh I see the throwing and catching in the same block. Will clean that up. If savedToken is null, isAuthenticationEnabled is tried next.

Reply:
Refactored the code